### PR TITLE
SQL2 - Dao part

### PR DESCRIPTION
**Pull request by AdiFinkelman**

### CODE REVIEW

File: src/test/kotlin/com/hibob/academy/dao/PetDaoTest.kt [hunk 1]
```diff
@@ -71,11 +72,49 @@ class PetDaoTest @Autowired constructor(private val sql: DSLContext) {
...
 
+    //sql 2 tests
+    @Test
+    fun `get pets by owner`() {
+        val ownerId = 2L
+        val petCreationRequest1 = PetCreationRequest("Tom", PetType.CAT, companyId, LocalDate.now(), ownerId)
+        val petCreationRequest2 = PetCreationRequest("Luke", PetType.DOG, companyId, LocalDate.now(), ownerId)
+        petDao.createNewPet(petCreationRequest1)
+        petDao.createNewPet(petCreationRequest2)
+        val expectedResult = listOf(petCreationRequest1, petCreationRequest2).map { it.name }
```
Comment (mayhuberman) [inline src/test/kotlin/com/hibob/academy/dao/PetDaoTest.kt:83]: better to check equals of the whole object and not only the name

File: src/test/kotlin/com/hibob/academy/dao/PetDaoTest.kt [hunk 2]
```diff
@@ -71,11 +72,49 @@ class PetDaoTest @Autowired constructor(private val sql: DSLContext) {
...
+        assertEquals(expectedResult, petDao.countPetsByType())
+    }
+
     @BeforeEach
     @AfterEach
     fun cleanup() {
         sql.deleteFrom(table)
             .where(table.companyId.eq(companyId))
+            .or(table.companyId.eq(testCompanyId))
             .execute()
```
Comment (mayhuberman) [inline src/test/kotlin/com/hibob/academy/dao/PetDaoTest.kt:118]: shouldn't do cleanup this way, add a delete method to the dao that expects companyId as a parameter

File: src/test/kotlin/com/hibob/academy/dao/PetDaoTest.kt [hunk 3]
```diff
@@ -71,11 +72,49 @@ class PetDaoTest @Autowired constructor(private val sql: DSLContext) {
...
+        val expectedResult = listOf(petCreationRequest1, petCreationRequest2).map { it.name }
+        assertEquals(expectedResult, petDao.getPetsByOwner(ownerId, companyId).map { it.name })
+    }
+
+    @Test
+    fun `get pets by owner when owner id are different`() {
+        val ownerId = 2L
+        petDao.createNewPet(pet1)
+        petDao.createNewPet(pet2)
+        val expectedResult = listOf(pet2).map { it.name }
```
Comment (mayhuberman) [inline src/test/kotlin/com/hibob/academy/dao/PetDaoTest.kt:92]: why only pet2? why is it different?

File: src/test/kotlin/com/hibob/academy/dao/PetDaoTest.kt (hunk 2 above)
Comment (AdiFinkelman) [reply src/test/kotlin/com/hibob/academy/dao/PetDaoTest.kt:118]: should I call this method in cleanup or in the test function?

File: src/test/kotlin/com/hibob/academy/dao/PetDaoTest.kt (hunk 3 above)
Comment (AdiFinkelman) [reply src/test/kotlin/com/hibob/academy/dao/PetDaoTest.kt:92]: The difference is the ownerId, pet1 has ownerId = 1 and pet2 has ownerId = 2.
When I call getPetsByOwner Im calling with ownerId = 2

File: src/main/kotlin/com/hibob/academy/dao/PetDao.kt [hunk 4]
```diff
@@ -54,4 +55,19 @@ class PetDao @Autowired constructor(private val sql: DSLContext) {
...
+
+    //sql 2
+    fun getPetsByOwner(ownerId: Long, companyId: Long): List<Pet> =
+        sql.select(petTable.id, petTable.name, petTable.type, petTable.companyId, petTable.dateOfArrival, petTable.ownerId)
+            .from(petTable)
+            .where(petTable.ownerId.eq(ownerId))
+            .and(petTable.companyId.eq(companyId))
+            .fetch(petMapper)
+
+    fun countPetsByType(): Map<String, Int> {
```
Comment (mayhuberman) [inline src/main/kotlin/com/hibob/academy/dao/PetDao.kt:67]: should return Map<PetType, Int>

**Review by mayhuberman — APPROVED** [review]

**Pull request merged**